include/boost/random/linear_congruential.hpp

```diff
@@ -137,7 +137,7 @@ class linear_congruential_engine
             _x = x0 % modulus;
         }
         // handle negative seeds
-        if(_x <= 0 && _x != 0) {
+        if(_x <= 0) {
```

Comment:
I think you simplified it wrong, it should be `_x < 0` (like the comment above says). Also, there is already a PR #64 for this.